EIPS/eip-1474.md

```diff
@@ -0,0 +1,2314 @@
+
+#### Parameters
+
+_(none)_
+
+#### Returns
+
+{[`Quantity`](#quantity)} - current gas price in wei
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_gasPrice",
+    "params": []
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": "0x09184e72a000"
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_getBalance</strong></code></summary>
+
+#### Description
+
+Returns the balance of an address in wei
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Data`](#data)}|address to query for balance|
+|2|{[`Quantity`](#quantity)\|`string`}|block number, or one of `"latest"`, `"earliest"` or `"pending"`|
+
+#### Returns
+
+{[`Quantity`](#quantity)} - balance of the provided account in wei
+
+#### Example
+
+```sh
+# Request
+curl -X POST --data '{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "method": "eth_getBalance",
+    "params": ["0xc94770007dda54cF92009BFF0dE90c06F603a09f", "latest"]
+}' <url>
+
+# Response
+{
+    "id": 1337,
+    "jsonrpc": "2.0",
+    "result": "0x0234c8a3397aab58"
+}
+```
+---
+</details>
+
+<details>
+<summary><code><strong>eth_getBlockByHash</strong></code></summary>
+
+#### Description
+
+Returns information about a block specified by hash
+
+#### Parameters
+
+|#|Type|Description|
+|-|-|-|
+|1|{[`Data`](#data)}|hash of a block|
+|2|{`boolean`}|`true` will pull full transaction objects, `false` will pull transaction hashes|
+
+#### Returns
+
+{`null|object`} - `null` if no block is found, otherwise a block object with the following members:
+
+- {[`Data`](#data)} `extraData` - "extra data" field of this block
+- {[`Data`](#data)} `hash` - block hash or `null` if pending
+- {[`Data`](#data)} `logsBloom` - logs bloom filter or `null` if pending
+- {[`Data`](#data)} `miner` - address that received this block's mining rewards
+- {[`Data`](#data)} `nonce` - proof-of-work hash or `null` if pending
+- {[`Data`](#data)} `parentHash` - parent block hash
+- {[`Data`](#data)} `receiptsRoot` -root of the this block's receipts trie
+- {[`Data`](#data)} `sha3Uncles` - SHA3 of the uncles data in this block
+- {[`Data`](#data)} `stateRoot` - root of this block's final state trie
+- {[`Data`](#data)} `transactionsRoot` - root of this block's transaction trie
+- {[`Quantity`](#quantity)} `difficulty` - difficulty for this block
+- {[`Quantity`](#quantity)} `gasLimit` - maximum gas allowed in this block
+- {[`Quantity`](#quantity)} `gasUsed` - total used gas by all transactions in this block
+- {[`Quantity`](#quantity)} `number` - block number or `null` if pending
+- {[`Quantity`](#quantity)} `size` - size of this block in bytes
+- {[`Quantity`](#quantity)} `timestamp` - unix timestamp of when this block was collated
+- {[`Quantity`](#quantity)} `totalDifficulty` - total difficulty of the chain until this block
+- {`Array<Transaction>`} `transactions` - list of transaction objects or hashes
```

Comment:
It's not clear what a Transaction object is, that's defined implicitly by the `eth_getTransactionBy*` sections. This EIP would be easier to read if the definition of Transaction was pulled out and explicitly named somewhere.